alacritty/src/input.rs

```diff
@@ -531,51 +566,12 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
             let display_offset = self.ctx.terminal().grid().display_offset();
             let point = self.ctx.mouse().point(&self.ctx.size_info(), display_offset);
 
-            match button {
-                MouseButton::Left => self.on_left_click(point),
-                MouseButton::Right => self.on_right_click(point),
-                // Do nothing when using buttons other than LMB.
-                _ => self.ctx.mouse_mut().click_state = ClickState::None,
+            if let MouseButton::Left = button {
+                self.on_left_click(point)
```

Comment:
Why did you remove the reset of the mouse state? That way pressing lmb -> rmb -> lmb will probably still trigger semantic selection. However it shouldn't. We need to reset the state after clicking to keep track of the buttons.

Reply:
Above when doing the double- and triple- click checks, the first check is to see if the mouse button has changed from the last click and if it did, it resets to `ClickState::Click`, so lmb -> rmb -> lmb will result in three single clicks.  I think that the test called `multi_click_separate_buttons` shows that this is the case. I also manually tested this with the released version as well as the new code and it works.

Reply:
Hm I think that makes some sense. It should work fine since the mouse click state update is done earlier already as you've said. If there is an issue with this, chances are that logic should be changed rather than branching here.